packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3092,21 +3088,30 @@ export class ContainerRuntime
 	private async fetchLatestSnapshotFromStorage(
 		logger: ITelemetryLogger,
 		event: ITelemetryGenericEvent,
-	): Promise<{ snapshotTree: ISnapshotTree; versionId: string }> {
+		readAndParseBlob: ReadAndParseBlob,
+		versionId?: string,
+	): Promise<{ snapshotTree: ISnapshotTree; versionId: string; latestSnapshotRefSeq: number }> {
 		return PerformanceEvent.timedExecAsync(
 			logger,
 			event,
 			async (perfEvent: {
 				end: (arg0: {
 					getVersionDuration?: number | undefined;
 					getSnapshotDuration?: number | undefined;
+					snapshotRefSeq?: number | undefined;
+					snapshotVersion?: string | undefined;
 				}) => void;
 			}) => {
-				const stats: { getVersionDuration?: number; getSnapshotDuration?: number } = {};
+				const stats: {
+					getVersionDuration?: number;
+					getSnapshotDuration?: number;
+					snapshotRefSeq?: number;
+					snapshotVersion?: string;
+				} = {};
 				const trace = Trace.start();
 
 				const versions = await this.storage.getVersions(
-					null,
+					versionId ?? null,
```

Comment:
>versionId ?? null, [](http://example.com/codeflow?start=5&length=18)

when using getVersions with versionId <> null, won't we get that specific version ? Do we want to always get the latest ?

Reply:
yeah, you are right. I also had the fix for the bug where we end up downloading snapshot for cache in this repo and while removing the code, this got left over :)

Reply:
Removed versionId